### PR TITLE
Add tests for socket_adapter

### PR DESCRIPTION
**Pull request by bonnetn**
Added integration tests for socket adapter:
- Spin a TCP server
- Connect the adapter to the server
- Make sure data is decoded properly

In order to be able to test the adapter fully, I needed to extract the asyncio.open_connection.

Now the adapter just knows about 'StreamReader'/'StreamWriter'.
The adapter logic was left unchanged.

NOTE: I also renamed it to SocketAdapter, since this adapter does nothing more than reading a socket. All the CAN logic is extracted in isotpcanserver.

I plan to leverage the exact same controller for receiving data from the LIDAR

### CODE REVIEW

File: src/main.py [hunk 1]
```diff
@@ -58,6 +62,15 @@
...
 
+async def _tcp_reader_writer(host: str,
+                             port: int) -> Tuple[StreamReader, StreamWriter]:
+    while True:
+        try:
+            return await asyncio.open_connection(host, port)
+        except ConnectionRefusedError:
+            LOGGER.get().error("connection_refused", port=port, host=host)
+
+
```
Comment (Melanie-Co) [inline src/main.py:73]: Why in main ?

File: src/main.py [hunk 2]
```diff
@@ -119,21 +132,17 @@ def _provide_fake_simulator_dependencies(i: DependencyContainer) -> None:
     i.provide('web_browser_client', WebBrowserClient)
     i.provide('replay_saver', ReplaySaver)
     i.provide('can_adapter', LoopbackCANAdapter)
-    i.provide('match_action_controller', MatchActionController)
-    i.provide('protobuf_handler', ProtobufHandler)
```
Comment (Melanie-Co) [inline src/main.py:123]: Why not have them in simulation mode ?

File: src/robot/adapter/socket/socket_adapter_test.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        # Two packets.
+        (b'<DEADBEEF>\n<CAFEBABE>\n',
+         [b'\xDE\xAD\xBE\xEF', b'\xCA\xFE\xBA\xBE']),
+
+        # Two packets without trailing newline.
+        (b'<DEADBEEF>\n<CAFEBABE>', [b'\xDE\xAD\xBE\xEF', b'\xCA\xFE\xBA\xBE']),
+
+        # Lots of newlines between packets.
+        (b'<01>\n\n\n\n<02>', [b'\x01', b'\x02']),
+    ])
```
Comment (Melanie-Co) [inline src/robot/adapter/socket/socket_adapter_test.py:67]: + tests for invalid packets ?

File: src/main.py [hunk 4]
```diff
@@ -164,14 +182,14 @@ async def main() -> None:
...
-    i = _get_container(is_simulation)
+    i = await _get_container(is_simulation)
 
     lidar_adapter: LIDARAdapter = i.get('lidar_adapter')
     lidar_controller: LidarController = i.get('lidar_controller')
     lidar_adapter.register_handler(lidar_controller.set_detection)
 
     can_adapter: CANAdapter = i.get('can_adapter')
-    socket_can_adapter: SocketCANAdapter = i.get('socket_can_adapter')
+    socket_can_adapter: SocketAdapter = i.get('socket_can_adapter')
```
Comment (Melanie-Co) [inline src/main.py:192]: If you also use it for the lidar don't call it socket_can_adapter.

File: src/main.py (hunk 1 above)
Comment (bonnetn) [reply src/main.py]: Good callout, moved to utils

File: src/main.py (hunk 2 above)
Comment (bonnetn) [reply src/main.py:123]: I moved them to the 'common' function, so they are instantiated both for simulation and production mode

File: src/robot/adapter/socket/socket_adapter_test.py (hunk 3 above)
Comment (bonnetn) [reply src/robot/adapter/socket/socket_adapter_test.py:67]: Done!

File: src/main.py (hunk 4 above)
Comment (bonnetn) [reply src/main.py]: Agreed & done

**Pull request merged**